python/src/spark_rapids_ml/classification.py

```diff
@@ -984,44 +990,85 @@ def coefficients(self) -> Vector:
         Model coefficients.
         """
-        assert len(self.coef_) == 1, "multi classes not supported yet"
-        return Vectors.dense(cast(list, self.coef_[0]))
+        if len(self.coef_) == 1:
+            return Vectors.dense(cast(list, self.coef_[0]))
+        else:
+            raise Exception(
+                "Multinomial models contain a matrix of coefficients, use coefficientMatrix instead."
+            )
 
     @property
     def intercept(self) -> float:
         """
         Model intercept.
         """
-        assert len(self.intercept_) == 1, "multi classes not supported yet"
-        return self.intercept_[0]
+        if len(self.intercept_) == 1:
+            return self.intercept_[0]
+        else:
+            raise Exception(
+                "Multinomial models contain a vector of intercepts, use interceptVector instead."
+            )
+
+    @property
+    def coefficientMatrix(self) -> Matrix:
+        """
+        Model coefficients.
+        """
+
+        n_rows = len(self.coef_)
+        n_cols = len(self.coef_[0])
+        flat_coef = [c for row in self.coef_ for c in row]
+        return DenseMatrix(
+            numRows=n_rows, numCols=n_cols, values=flat_coef, isTransposed=True
+        )
+
+    @property
+    def interceptVector(self) -> Vector:
+        """
+        Model intercept.
+        """
+        return Vectors.dense(cast(list, self.intercept_))
+
+    @property
+    def numClasses(self) -> int:
+        return self.num_classes
 
     def _get_cuml_transform_func(
         self, dataset: DataFrame, category: str = transform_evaluate.transform
     ) -> Tuple[_ConstructFunc, _TransformFunc, Optional[_EvaluateFunc],]:
         coef_ = self.coef_
         intercept_ = self.intercept_
+        classes_ = self.classes_
         n_cols = self.n_cols
         dtype = self.dtype
 
         def _construct_lr() -> CumlT:
+            import cupy as cp
             import numpy as np
-            from cuml.internals.input_utils import input_to_cuml_array
+            from cuml.internals.global_settings import GlobalSettings
+            from cuml.internals.input_utils import CumlArray, input_to_cuml_array
             from cuml.linear_model.logistic_regression_mg import LogisticRegressionMG
 
             lr = LogisticRegressionMG(output_type="numpy")
             lr.n_cols = n_cols
             lr.dtype = np.dtype(dtype)
-            lr.intercept_ = input_to_cuml_array(
-                np.array(intercept_, order="C").astype(dtype)
-            ).array
-            lr.coef_ = input_to_cuml_array(
-                np.array(coef_, order="C").astype(dtype)
-            ).array
-            # TBD: infer class indices from data for > 2 classes
-            # needed for predict_proba
+
+            gpu_intercept_ = cp.array(intercept_, order="C", dtype=dtype)
+            gpu_coef_ = cp.array(coef_, order="F", dtype=dtype).T
+            gpu_stacked = GlobalSettings().xpy.vstack([gpu_coef_, gpu_intercept_])
```

Comment:
Does cupy not have vstack?

Reply:
Revised to use cupy. Just found it does have vstack.